src/main/java/com/mastery/java/task/controllers/EmployeeController.java

```diff
@@ -0,0 +1,157 @@
+    }
+
+    @PostMapping("/add")
+    public String employeeAdd(@Valid Employee employee, BindingResult bindingResult, Model model) throws DuplicateEmployeeException, InvalidDateException {
+        if (bindingResult.hasErrors()) {
+            log.info("Invalid employee has " + bindingResult.getFieldErrorCount() + " error");
+            log.trace("Invalid employee: " + bindingResult.getAllErrors());
+            logCon.info("Invalid employee has " + bindingResult.getFieldErrorCount() + " error");
+            model.addAttribute("employee", employee);
+            return "employee-add";
+        }
+        log.info("Employee added: " + employee);
+        logCon.info("Employee added: " + employee);
+        employeeService.save(employee);
+        return "redirect:/";
+    }
+
+    @GetMapping("/get/{id}")
+    public String employeeDetails(@PathVariable(value = "id") int id, Model model) {
+        Employee employee = employeeService.getById(id);
+        log.info("Employee with ID " + id + " received: " + employee);
+        logCon.info("Employee with ID " + id + " received: " + employee);
+        model.addAttribute("employee", employee);
+        return "employee-details";
+    }
+
+    @PostMapping("/{id}/delete")
+    public String employeeDelete(@PathVariable(value = "id") int id) throws InterruptedException {
+        this.jmsMessagingTemplate.convertAndSend(this.queue, String.valueOf(id));
+        log.info("Employee with ID " + id + " deleted");
+        logCon.info("Employee with ID " + id + " deleted");
+        Thread.sleep(50);
+        return "redirect:/";
+    }
+
+    @GetMapping("/{id}/edit")
+    public String employeeEdit(@PathVariable(value = "id") int id, Model model) {
+        Employee employee = employeeService.getById(id);
+        logCon.trace("Go to the page for change an employee with id " + id);
+        model.addAttribute("employee", employee);
+        return "employee-edit";
+    }
+
+    @PostMapping("/{id}/edit")
+    public String employeeUpdate(@PathVariable(value = "id") int id,
+                                 @Valid Employee employee, BindingResult bindingResult, Model model) {
+        employee.setEmployeeId(id);
+        if (bindingResult.hasErrors()) {
+            model.addAttribute("employee", employee);
+            log.warn("Invalid employee has " + bindingResult.getFieldErrorCount() + " error");
+            log.debug("Invalid employee: " + bindingResult.getAllErrors());
+            logCon.warn("Invalid employee has " + bindingResult.getFieldErrorCount() + " error");
+            return "employee-edit";
+        }
+        employeeService.update(employee);
+        log.info("Employee with ID " + id + " changed: " + employee);
+        logCon.info("Employee with ID " + id + " changed: " + employee);
+        return "redirect:/";
+    }
+
+    @PostMapping("/lvl/{level}")
+    public String changeLogLevel(@PathVariable(value = "level") String level) throws IOException {
```

Comment:
I believe the logger level might be changed this way, but I think it's preferred to use Spring Boot actuator for this

Reply:
This method needed for changing level Logger from ui website, in this method call actuator, and it changes the level
the method collects a request with the body to url - "http://localhost:8080/actuator/loggers/com.mastery.java.task"

Reply:
Ok, got you